test/topology/test_random_tables.py

```diff
@@ -3,16 +3,23 @@
 #
 # SPDX-License-Identifier: AGPL-3.0-or-later
 #
+import time
 import pytest
 from cassandra.protocol import InvalidRequest, ReadFailure                         # type: ignore
 
+from test.topology.util import wait_for_token_ring_and_group0_consistency
+
 
 # Simple test of schema helper
 @pytest.mark.asyncio
 async def test_new_table(manager, random_tables):
     cql = manager.cql
     assert cql is not None
     table = await random_tables.add_table(ncolumns=5)
+    # Before performing data queries, make sure that the token ring
+    # has converged (we're using ring_delay = 0).
+    # Otherwise the queries may pick wrong replicas.
+    await wait_for_token_ring_and_group0_consistency(manager, time.time() + 60)
```

Comment:
Shan't we add this to topology operations in the manager, not to a specific test?
I'm talking add/remove node.

Reply:
I see some reasons for not doing it:
1.  not all tests need it because not all tests do data queries, it will make those tests unnecessarily longer (waiting for gossip convergence)
2. it shouldn't be necessary once we switch to raft-based topology, and better to not have it once we do (we should then test what happens when we do operations without any waiting)